sklearn/linear_model/least_angle.py

```diff
@@ -1078,7 +1071,13 @@ def fit(self, X, y):
         # init cross-validation generator
         cv = check_cv(self.cv, X, y, classifier=False)
 
-        Gram = 'auto' if self.precompute else None
+        # As we use cross-validation, the Gram matrix is not precomputed here
```

Comment:
I would say raise an error instead of a warning but it depends on our user nursing level